gin/middleware.go

```diff
@@ -9,6 +9,18 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+// New returns a function that satisfies gin.HandlerFunc interface
+func New(engine *gin.Engine, notifier *gobrake.Notifier) func(c *gin.Context) {
```

Comment:
Why do we need to introduce `New()`?

Reply:
We will be standardising this function across all the middleware like apexlog, gin, echo etc.